Objects/unicodeobject.c

```diff
@@ -13362,6 +13476,18 @@ _PyUnicodeWriter_WriteASCIIString(_PyUnicodeWriter *writer,
     return 0;
 }
 
+int
+PyUnicodeWriter_WriteUTF8(PyUnicodeWriter *writer,
+                          const char *str,
+                          Py_ssize_t size)
+{
+    if (size == -1) {
```

Comment:
We should never call `unicode_decode_utf8_writer` with a negative size.

```suggestion
    if (size < 0) {
```